File: services/delivery/src/main/java/io/wkrzywiec/fooddelivery/delivery/domain/outgoing/FoodInPreparation.java
package io.wkrzywiec.fooddelivery.delivery.domain.outgoing;

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody;

import java.util.UUID;

public record FoodInPreparation(UUID orderId, int version) implements IntegrationMessageBody {
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/incoming/FoodInPreparation.kt
package io.wkrzywiec.fooddelivery.ordering.domain.incoming

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody
import java.util.*

data class FoodInPreparation(val id: UUID, val aggregateVersion: Int) : IntegrationMessageBody {
    override fun orderId(): UUID = id
    override fun version(): Int = aggregateVersion
}
